Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m

```diff
@@ -1144,13 +1145,24 @@ - (CGSize)bodyMediaSizeForContentWidth:(int)contentWidth
 
             const CGFloat maxMediaWidth = maxMessageWidth;
             const CGFloat maxMediaHeight = maxMessageWidth;
-            CGFloat mediaWidth = (CGFloat)round(maxMediaHeight * contentAspectRatio);
-            CGFloat mediaHeight = (CGFloat)round(maxMediaHeight);
+            CGFloat mediaWidth = maxMediaHeight * contentAspectRatio;
+            CGFloat mediaHeight = maxMediaHeight;
             if (mediaWidth > maxMediaWidth) {
-                mediaWidth = (CGFloat)round(maxMediaWidth);
-                mediaHeight = (CGFloat)round(maxMediaWidth / contentAspectRatio);
+                mediaWidth = maxMediaWidth;
+                mediaHeight = maxMediaWidth / contentAspectRatio;
             }
-            return CGSizeMake(mediaWidth, mediaHeight);
+
+            // We don't want to blow up small images unnecessarily.
```

Comment:
I'd like to discuss this with Signal-iOS team and Myles.  Essentially, if you send a 1x1 png, we blow it up to a huge PNG.  I think we should blow it up, but within limits.